Other/MergeShops/Adam1a1Merge.cs

```diff
@@ -0,0 +1,152 @@
+using System.Diagnostics;
+using System.Reflection.PortableExecutable;
+//cs_include Scripts/CoreBots.cs
+//cs_include Scripts/CoreFarms.cs
+//cs_include Scripts/CoreAdvanced.cs
+using Skua.Core.Interfaces;
+using Skua.Core.Models.Items;
+using Skua.Core.Options;
+
+public class Adam1a1Merge
+{
+    private IScriptInterface Bot => IScriptInterface.Instance;
+    private CoreBots Core => CoreBots.Instance;
+    private CoreFarms Farm = new();
+    private CoreAdvanced Adv = new();
+    private static CoreAdvanced sAdv = new();
+
+    public List<IOption> Generic = sAdv.MergeOptions;
+    public string[] MultiOptions = { "Generic", "Select" };
+    public string OptionsStorage = sAdv.OptionsStorage;
+    // [Can Change] This should only be changed by the author.
+    //              If true, it will not stop the script if the default case triggers and the user chose to only get mats
+    private bool dontStopMissingIng = false;
+
+    public void ScriptMain(IScriptInterface Bot)
+    {
+        Core.BankingBlackList.AddRange(new[] { "Diabolical Ectomancer", "Fresh Ectoplasm", "IOU Slip", "EctoBlade", "Ectoplasmic Chains", "Bongo Cart Pet", "Reho's Golden Sword Hilt"});
+        Core.SetOptions();
+
+        BuyAllMerge();
+
+        Core.SetOptions(false);
+    }
+
+    public void BuyAllMerge(string buyOnlyThis = null, mergeOptionsEnum? buyMode = null)
+    {
+        //Only edit the map and shopID here
+        Adv.StartBuyAllMerge("battleunderb", 1990, findIngredients, buyOnlyThis, buyMode: buyMode);
+
+        #region Dont edit this part
+        void findIngredients()
+        {
+            ItemBase req = Adv.externalItem;
+            int quant = Adv.externalQuant;
+            int currentQuant = req.Temp ? Bot.TempInv.GetQuantity(req.Name) : Bot.Inventory.GetQuantity(req.Name);
+            if (req == null)
+            {
+                Core.Logger("req is NULL");
+                return;
+            }
+
+            switch (req.Name)
+            {
+                default:
+                    bool shouldStop = Adv.matsOnly ? !dontStopMissingIng : true;
+                    Core.Logger($"The bot hasn't been taught how to get {req.Name}." + (shouldStop ? " Please report the issue." : " Skipping"), messageBox: shouldStop, stopBot: shouldStop);
+                    break;
+                #endregion
+
+                case "Diabolical Ectomancer":
+                    Core.FarmingLogger(req.Name, quant);
+                    Core.EquipClass(ClassType.Farm);
+                    Core.RegisterQuests(7880);
+                    while (!Bot.ShouldExit && !Core.CheckInventory(req.Name, quant))
+                    {
+                        Core.HuntMonster("banished", "r14", "Left", "*", "Desterrat Moya", "Infected Tentacle", 1, log: false);
+                        Bot.Wait.ForPickup(req.Name);
+                    }
+                    Core.CancelRegisteredQuests();
+                    break;
+
+                case "Fresh Ectoplasm":
```

Comment:
multiple cases of same item, figure it out 👍  only combine them like these if *ALL* the items come from the same while (buncha hunts together). otherwise keep them seperate.